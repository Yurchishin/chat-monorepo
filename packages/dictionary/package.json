{
  "name": "@monorepo/dictionary",
  "version": "0.0.1",
  "description": "",
  "keywords": [],
  "author": "Yurii Rybak <yurarary@gmail.com>",
  "license": "ISC",
  "main": "dist/index.js",
  "types": "dist/index.d.ts",
  "publishConfig": {
    "access": "public"
  },
  "scripts": {
    "build": "tsc -p tsconfig.json",
    "watch": "tsc -p tsconfig.json --watch",
    "start": "node dist/index.js",
    "start:build": "npm run build & npm run start",
    "lint": "npm run lint:eslint npm run lint:tslint",
    "lint:fix": "npm run lint:eslint-fix npm run lint:tslint-fix",
    "lint:eslint": "eslint --ext .ts src",
    "lint:eslint-fix": "eslint --fix --ext .ts src",
    "lint:tslint": "tslint -c ./tslint.json -p ./tsconfig.json",
    "lint:tslint-fix": "tslint -c ./tslint.json -p ./tsconfig.json --fix"
  },
  "devDependencies": {
    "typescript": "^3.7.5"
  },
  "dependencies": {
    "@types/node": "^13.1.8"
  }
}
