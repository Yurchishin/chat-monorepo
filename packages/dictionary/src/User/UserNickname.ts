export const UserNicknameRange = {
    MIN: 5,
    MAX: 20,
}

export const UserNicknameRegExp = new RegExp('^[^0-9][^@#_-]+$')

export const UserNicknameError = {
    TYPE: 'Wrong nickname type',
    MIN_LENGTH: 'Wrong nickname type min length',
    MAX_LENGTH: 'Wrong nickname type max length',
    REG_EXP: 'Wrong nickname format',
}
