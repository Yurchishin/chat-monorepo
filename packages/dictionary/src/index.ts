export * from './Hosts'
export * from './NodeEnv'
export const Hui = 'hi'
