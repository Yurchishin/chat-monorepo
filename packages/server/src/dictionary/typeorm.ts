export const TYPEORM_CONNECTION = 'postgres'
export const TYPEORM_ENTITIES = 'dist/entity/**/*.js'
export const TYPEORM_MIGRATIONS = 'dist/migration/**/*.js'
export const TYPEORM_ENTITIES_DIR = 'src/entity'
export const TYPEORM_MIGRATIONS_DIR = 'src/migration'

