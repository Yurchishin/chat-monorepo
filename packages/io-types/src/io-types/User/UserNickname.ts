import * as t from 'io-ts'
import { UserNicknameError, UserNicknameRange, UserNicknameRegExp } from '@monorepo/dictionary'
import { withMessage } from 'io-ts-types/lib/withMessage'
import tString from '../String'

export const userNickname = t.intersection([
        withMessage(tString.string, () => UserNicknameError.TYPE),
        withMessage(tString.lessThenOrEqual(UserNicknameRange.MAX), () => UserNicknameError.MAX_LENGTH),
        withMessage(tString.greaterThenOrEqual(UserNicknameRange.MIN), () => UserNicknameError.MIN_LENGTH),
        withMessage(tString.regExp(new RegExp(UserNicknameRegExp)), () => UserNicknameError.REG_EXP),
], 'UserNickname')
