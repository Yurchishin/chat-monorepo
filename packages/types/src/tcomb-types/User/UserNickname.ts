import t, { Intersection } from 'tcomb-validation'
import { UserNicknameError, UserNicknameRange, UserNicknameRegExp } from '@monorepo/dictionary'
import { addErrorMessage } from '../../utils'
import tString from '../String'

export const UserNickname: Intersection<string> = t.intersection([
        addErrorMessage(tString.String, UserNicknameError.TYPE),
        addErrorMessage(tString.StringLTE(UserNicknameRange.MAX), UserNicknameError.MAX_LENGTH),
        addErrorMessage(tString.StringGTE(UserNicknameRange.MIN), UserNicknameError.MIN_LENGTH),
        addErrorMessage(tString.StringRegExp(UserNicknameRegExp), UserNicknameError.REG_EXP),
], 'UserNickname')
