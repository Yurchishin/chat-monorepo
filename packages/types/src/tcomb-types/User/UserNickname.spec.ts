import {UserNickname} from './UserNickname'
import {expectTypeSuccess, expectTypeErrors} from "../../tests"
import { UserNicknameError } from "@monorepo/dictionary";

describe('UserNickname', () => {
	expectTypeSuccess(UserNickname, 'yuryy')
	expectTypeSuccess(UserNickname, 'yury12')
	expectTypeSuccess(UserNickname, 'yury12')
	expectTypeErrors(UserNickname, 'yuryyuryyuryyuryyuryyury', [
		UserNicknameError.MAX_LENGTH,
	])
	expectTypeErrors(UserNickname, 'yury', [
		UserNicknameError.MIN_LENGTH,
	])
	expectTypeErrors(UserNickname, 1, [
		UserNicknameError.TYPE,
		UserNicknameError.MAX_LENGTH,
		UserNicknameError.MIN_LENGTH,
		UserNicknameError.REG_EXP,
	])
})
