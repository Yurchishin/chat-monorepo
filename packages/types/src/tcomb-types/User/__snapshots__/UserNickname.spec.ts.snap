// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserNickname expectTypeFailure(1): TYPE_FAILURE::UserNickname(1) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong nickname type",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong nickname type max length",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong nickname type min length",
      "path": Array [],
    },
    Struct {
      "actual": 1,
      "expected": [Function],
      "message": "Wrong nickname format",
      "path": Array [],
    },
  ],
  "value": 1,
}
`;

exports[`UserNickname expectTypeFailure(yury): TYPE_FAILURE::UserNickname(yury) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "yury",
      "expected": [Function],
      "message": "Wrong nickname type min length",
      "path": Array [],
    },
  ],
  "value": "yury",
}
`;

exports[`UserNickname expectTypeFailure(yuryyuryyuryyuryyuryyury): TYPE_FAILURE::UserNickname(yuryyuryyuryyuryyuryyury) 1`] = `
Struct {
  "errors": Array [
    Struct {
      "actual": "yuryyuryyuryyuryyuryyury",
      "expected": [Function],
      "message": "Wrong nickname type max length",
      "path": Array [],
    },
  ],
  "value": "yuryyuryyuryyuryyuryyury",
}
`;
